Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.cpp

```diff
@@ -17,50 +17,62 @@ SplitAndScatterFunc::SplitAndScatterFunc (const std::string& collision_name,
     {
         const amrex::ParmParse pp_collision_name(collision_name);
 
-        // Check if ionization is one of the scattering processes by querying
-        // for any specified product species (ionization is the only current
-        // DSMC process with products)
-        amrex::Vector<std::string> product_species;
-        pp_collision_name.queryarr("product_species", product_species);
+        // Check if the scattering processes include reactions that produce macroparticles in new species
+        // (i.e. not in the incident species list), i.e. if it contains ionization or charge exchange
+        amrex::Vector<std::string> scattering_processes;
+        pp_collision_name.queryarr("scattering_processes", scattering_processes);
+        const bool reaction_produces_new_species = std::any_of(scattering_processes.begin(), scattering_processes.end(), [](const std::string& process) {
+            return process == "ionization" || process == "charge_exchange";
+        });
```

Comment:
We should think about which processes should be in the constructor of DSMCFunc vs SplitAndScatterFunc